src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -346,15 +346,33 @@ export class MeetingManager implements AudioVideoObserver {
     );
   }
 
-  async listAndSelectDevices(): Promise<void> {
+  async listAndSelectDevices(deviceLabels: DeviceLabels | DeviceLabelTrigger = DeviceLabels.AudioAndVideo): Promise<void> {
     await this.updateDeviceLists();
 
-    const hasAudioInput = this.audioInputDevices?.some(value => value.label) || false;
-    const hasAudioOutput = this.audioOutputDevices?.some(value => value.label) || false;
-    const hasVideoInput = this.videoInputDevices?.some(value => value.label) || false;
+    // If `deviceLabels` is of `DeviceLabelTrigger` type, no device will be selected.
+    // In this case, you need to handle the device selection yourself.
+    if (typeof deviceLabels === 'function') return;
+
+    let isAudioDevicesRequested: boolean = false;
+    let isVideoDevicesRequested: boolean = false;
+
+    switch (deviceLabels) {
+      case DeviceLabels.None:
+        break;
```

Comment:
For the readability, I keep this empty case here to cover all the cases. Not sure if it's reasonable to do this.

Reply:
Closed/exhaustive sets are best — this is good Rust style, so I support it here!